server/command/install.go

```diff
@@ -11,28 +11,55 @@ import (
 
 	"github.com/mattermost/mattermost-plugin-apps/apps"
 	"github.com/mattermost/mattermost-plugin-apps/server/httpin/dialog"
-	"github.com/mattermost/mattermost-plugin-apps/server/utils"
 )
 
 func (s *service) executeInstall(params *params) (*model.CommandResponse, error) {
 	appSecret := ""
+	manifestURL := ""
 	fs := pflag.NewFlagSet("", pflag.ContinueOnError)
 	fs.StringVar(&appSecret, "app-secret", "", "App secret")
-	if err := fs.Parse(params.current); err != nil {
+	fs.StringVar(&manifestURL, "url", "", "App's manifest URL")
+	err := fs.Parse(params.current)
+	if err != nil {
 		return errorOut(params, err)
 	}
 
-	if !s.mm.User.HasPermissionTo(params.commandArgs.UserId, model.PERMISSION_MANAGE_SYSTEM) {
-		return errorOut(params, utils.ErrForbidden)
-	}
+	var m *apps.Manifest
+	var appID apps.AppID
+	conf := s.conf.GetConfig()
+	if conf.MattermostCloudMode {
+		if len(params.current) == 0 {
+			return errorOut(params, errors.New("you must specify the app id"))
+		}
+		appID = apps.AppID(params.current[0])
+	} else {
+		switch {
+		case len(params.current) > 0:
```

Comment:
Does this also catch cases like `/apps install --url  http://localhost:3000/manifest`?